File: Nave_10_02_24_Datos_Propios.py
from openai import AzureOpenAI
from dotenv import load_dotenv
import os

load_dotenv()
azureOpenAI_endpoint = os.environ.get("AZUREOPENAI_ENDPOINT")
azureOpenAI_key = os.environ.get("AZUREOPENAI_KEY")
deployment_name = os.environ.get("DEPLOYMENT_NAME")
azureOpenAI_version = os.environ.get("AZUREOPENAI_VERSION")
search_endpoint = os.environ.get("SEARCH_ENDPOINT")
search_key = os.environ.get("SEARCH_KEY")
search_index_name = os.environ.get("SEARCH_INDEX_NAME")

client = AzureOpenAI(
    base_url=f'{azureOpenAI_endpoint}/openai/deployments/{deployment_name}/extensions',
    api_key=azureOpenAI_key,
    api_version= azureOpenAI_version,
)
question= 'que es la universidad de la habana?'
print(f"User: {question}")
completion = client.chat.completions.create(model= deployment_name, messages= [
    {
        "role": "user","content": question
        }
        ], extra_body={
            "dataSources": [
                {
                    "type": "AzureCognitiveSearch",
                    "parameters": {
                        "endpoint": search_endpoint,
                        "key": search_key,
                        "indexName": search_index_name,
                        "inScope": False,
                        "roleInformation": 'You are an AI assistant that helps people find information.',
                        "filter": None,
                        "strictness": 4,
                        "topNDocuments": 5,
                        }
                    }
                ], 
            "temperature":0,
            "top_p":1,
            "max_tokens":800, 
            "stop": None 
            })
# print(completion.model_dump_json(indent=2)) #json del response
print(f'Chatbot: {completion.choices[0].message.content}')

File: Nave_10_02_24_Datos_Propios.js
const { OpenAIClient,AzureKeyCredential } = require("@azure/openai");

const endpoint = "https://lanave-assistant-ia-service.openai.azure.com/";
const azureApiKey = "f93d8e26153543d18b2e1124ade3d01d";
const searchEndpoint = "https://lanaveassistantsearch.search.windows.net";
const searchKey = "gCXGhPxaufktNMDObsMReXknUhZTu5mpDQP1VZnDdvAzSeBOR61v";
const searchIndex = "lanave-assistant-index-dev"
const deploymentId = "gpt-4"

async function main(){
    const client = new OpenAIClient(endpoint, new AzureKeyCredential(azureApiKey));
    const messages = [
        { role: "user", content: "que es la nave?" },
    ];

    console.log(`User: ${messages.map((m) => m.content).join("\n")}`);

    const events = await client.streamChatCompletions(deploymentId, messages, {   
    azureExtensionOptions: {
      extensions: [
        {
          type: "AzureCognitiveSearch",
          endpoint: searchEndpoint,
          key: searchKey,
          indexName: searchIndex,
          inScope: false,
          roleInformation: "You are an AI assistant that helps people find information.",
          filter: undefined,
          strictness: 4,
          topNDocuments: 5,          
        },
      ],
    },
    temperature:0,
    top_p:1,
    max_tokens:800, 
    stop: undefined,  
  });
  let cadena=''
  for await (const event of events) {
    for (const choice of event.choices) {
      const delta = choice.delta?.content;
      if (delta !== undefined) {
        cadena += delta
      }
    }
  }
  console.log(`Chatbot: ${cadena}`)
}

main().catch((err) => {
  console.error("The sample encountered an error:", err);
});